sdk/github.md

```diff
@@ -83,14 +83,43 @@ This is the name of the file within the `.github/workflows` folder.
   - `assemble`:
     - build the project (archives / artifacts)
     - for Gradle Java projects this is `./gradlew assemble`
-  - `check` - for verification tasks, including:
+    - pushes artifacts, where logical, to `sdk.ably.com` (staging)
+  - `check` - for basic, quick verification tasks, including:
     - run linters
     - perform static analysis if available
     - run unit tests
     - generate code coverage reports from unit test runs if possible (probably uploaded as [artifacts](https://docs.github.com/en/actions/advanced-guides/storing-workflow-data-as-artifacts))
     - for Gradle Java projects this is `./gradlew check`
+    - should not run time consuming integration tests
+    - examples:
+      [ably-asset-tracking-android](https://github.com/ably/ably-asset-tracking-android/blob/main/.github/workflows/check.yml),
+      [ably-java](https://github.com/ably/ably-java/blob/main/.github/workflows/check.yml),
+      [ably-js](https://github.com/ably/ably-js/blob/main/.github/workflows/check.yml),
+      [engineering](https://github.com/ably/engineering/blob/main/.github/workflows/check.yaml)
+  - `docs`:
+    - build API reference documentation from interface commentary
+    - should push artifacts to `sdk.ably.com` (staging)
+    - examples:
+      [ably-asset-tracking-android](https://github.com/ably/ably-asset-tracking-android/blob/main/.github/workflows/docs.yml),
+      [ably-asset-tracking-swift](https://github.com/ably/ably-asset-tracking-swift/blob/main/.github/workflows/docs.yml),
+      [ably-flutter](https://github.com/ably/ably-flutter/blob/main/.github/workflows/docs.yml)
   - `emulate`:
+    - runs tests, typically time consuming integration tests
+    - spawns an emulator to do the testing - typically Android or iOS
     - for Gradle Java projects this is `./gradlew connectedCheck`
+    - example:
+      [ably-asset-tracking-android](https://github.com/ably/ably-asset-tracking-android/blob/main/.github/workflows/emulate.yml)
+  - `integration-test`:
+    - run time consuming integration tests
+    - example:
+      [ably-java](https://github.com/ably/ably-java/blob/main/.github/workflows/integration-test.yml)
+  - `publish`:
```

Comment:
I have [a workflow](https://github.com/ably/ably-js/blob/main/.github/workflows/bundle-report.yml) in ably-js which generates a [bundle size report](https://sdk.ably.com/builds/ably/ably-js/main/bundle-report/) using [source-map-explorer](https://www.npmjs.com/package/source-map-explorer). Looking through here I'm not entirely sure where this workflow would go? You could argue that it would make sense to go in publish or check, but I'm not sure that either are entirely appropriate (maybe I'm missing something though). It might be worth manually checking through all of the currently existing workflows in our SDKs and making sure that every single one has a clear and obvious place within these guidelines?

Reply:
Indeed, that was my intention... (1) establish guidelines; (2) conform what can conformed to fit; (3) list out what couldn't confirm, then back to step (1) to refine.

This could be an example where the verb `report` would suit? i.e. your `bundle-report.yml` will become `report.yaml`. Or another name. Or not. 😉 

I could address that in this PR.

Reply:
`report.yaml` sounds good to me. Especially with the other suggestion it could be `report:bundle-size.yaml` which would be nice. I'll resolve this now, am happy for you to add it to this PR or defer for later.

Reply:
https://github.com/ably/engineering/pull/13/commits/837a20fc6c70d49fc9594790e3bed2ade9cd766e